xbmc/GUIPassword.cpp

```diff
@@ -544,6 +545,14 @@ bool CGUIPassword::IsDatabasePathUnlocked(const std::string& strPath, VECSOURCES
   if (g_passwordManager.bMasterUser || profileManager->GetMasterProfile().getLockMode() == LOCK_MODE_EVERYONE)
     return true;
 
+  // Don't check plugin paths as they don't have an entry in sources.xml
+  // Base locked status on videos being locked and being in the video
+  // navigation screen (must have passed lock by then)
+  CURL url{strPath};
+  if (url.IsProtocol("plugin"))
+    return !(profileManager->GetCurrentProfile().videoLocked() &&
+             CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow() != WINDOW_VIDEO_NAV);
```

Comment:
Is there any particular reason this is only "video"? I would assume users may want to use locking for audio (or pictures, or whatever other plugin types we have) as well possibly?

This looks to be at a level where it shouldnt explicitly handle one specific media type

Reply:
> Is there any particular reason this is only "video"? I would assume users may want to use locking for audio (or pictures, or whatever other plugin types we have) as well possibly?
> 
> This looks to be at a level where it shouldnt explicitly handle one specific media type

Interestingly there is no lock on TV shows.
I've added a commit for music.
Jellyfin doesn't serve anything else I don't think.
I think the other plugins make content available through the addon itself and not the default kodi menu.

Reply:
Ignore the above. I've done more testing today.
Music doesn't go through `CGUIPassword::IsDatabasePathUnlocked` and works as it should do already.
TV shows and videos are part of `videodb` so there doesn't need to be a seperate lock for TV shows.
I think the only code that's needed is the original code.